sycl/doc/SYCL_compiler_and_runtime_design.md

```diff
@@ -162,6 +162,17 @@ number of device compilers for targets specified in the `-fsycl-targets`
 option.  If this option is not specified, then single SPIR-V target is assumed,
 and single device compiler for this target is invoked.
 
+In the driver, the following bools are defined to determine the compilation
+mode in SYCL:
+
+* IsSYCL : True if the user has passed `-fsycl` or `-sycl` to the compilation
```

Comment:
syntax of device only option is '--sycl'

Reply:
Let's document `-fsycl-is-device`.
We are going to remove `--sycl`.

Reply:
> Let's document `-fsycl-is-device`.
> We are going to remove `--sycl`.

I believe the option is '-fsycl-device-only', right?

Reply:
@mdtoguchi, right. I just realized that this renaming is not done yet.
Please, ignore my comment.

Reply:
Which one should I keep on my commit then?

Reply:
Please use '--sycl'

Reply:
> Let's document `-fsycl-is-device`.
> We are going to remove `--sycl`.

As an aside, would removal of the --sycl flag stop the ability to compile just the kernel bitcode and integration header or are you planning to add some other functionality/way of achieving the same thing with a different combination of flags? I ask as being able to separate the offloading and just compile the required device output has been quite useful. 

And if -save-temps is still broken (https://github.com/intel/llvm/issues/25) I imagine (but could be wrong) using the --sycl flag is currently the simplest way to get access to just the kernel bitcode and integration header if you want to inspect them. I'm not against removing it, but I have found the --sycl option useful in the past!

Sorry for derailing the code review.

Reply:
--sycl behavior is not going away, we are replacing the option with a more friendly external option name (-fsycl-device-only)

Reply:
Ah, great news. Thanks for the information!

Reply:
Pushed the fix , thanks!